cfme/tests/infrastructure/test_host.py

```diff
@@ -350,16 +352,22 @@ def test_hosts_not_displayed_several_times(appliance, provider, setup_provider):
 
 
 @pytest.fixture
-def setup_provider_min_hosts(request, provider, min_hosts=2):
-    num_hosts = len(provider.data.get('hosts', {}))
-    if num_hosts < min_hosts:
-        pytest.skip(f'Not enough hosts({num_hosts}) to run test. Need at least {min_hosts}')
+def setup_provider_min_hosts(request, appliance, provider, needed_hosts):
+    hosts_yaml = len(provider.data.get('hosts', {}))
+    if hosts_yaml < needed_hosts:
+        pytest.skip(f'Number of hosts defined in yaml for {provider} does not meet minimum '
+                    f'for test parameter {needed_hosts}, skipping and not setting up provider')
     # Function-scoped fixture to set up a provider
-    return setup_or_skip(request, provider)
+    if appliance.rest_api.collections.hosts.count < needed_hosts:  # This is total appliance
```

Comment:
This would need to be checked after `setup_or_skip` - but as discussed offline, an even better option is to use `len(provider.mgmt.list_host())

Reply:
Done